process/smartContract/hooks/blockChainHook.go

```diff
@@ -64,14 +69,23 @@ func NewBlockChainHookImpl(
 	}
 
 	blockChainHookImpl := &BlockChainHookImpl{
-		accounts:         args.Accounts,
-		pubkeyConv:       args.PubkeyConv,
-		storageService:   args.StorageService,
-		blockChain:       args.BlockChain,
-		shardCoordinator: args.ShardCoordinator,
-		marshalizer:      args.Marshalizer,
-		uint64Converter:  args.Uint64Converter,
-		builtInFunctions: args.BuiltInFunctions,
+		accounts:          args.Accounts,
+		pubkeyConv:        args.PubkeyConv,
+		storageService:    args.StorageService,
+		blockChain:        args.BlockChain,
+		shardCoordinator:  args.ShardCoordinator,
+		marshalizer:       args.Marshalizer,
+		uint64Converter:   args.Uint64Converter,
+		builtInFunctions:  args.BuiltInFunctions,
+		compiledScPool:    args.DataPool.SmartContracts(),
+		compiledScStorage: args.StorageService.GetStorer(dataRetriever.SmartContractUnit),
+	}
+
+	if check.IfNil(blockChainHookImpl.compiledScStorage) {
```

Comment:
you could move these checks before assigning the values to the `blockChainHookImpl`

Reply:
it is the same. It is less code like this.